src/structural_pipeline/COMMIT-Filter/commit_filter.m

```diff
@@ -0,0 +1,200 @@
+filteredFiberFile = configParams.commit_filter.filteredFiberFile;
+filteredFiberPropertiesFile = configParams.commit_filter.filteredFiberPropertiesFile;
+schemeFile = configParams.commit_filter.schemeFile;
+intermediateConnectomeFile = configParams.commit_filter.intermediateConnectomeFile;
+pythonInterpreter = configParams.commit_filter.pythonInterpreter;
+commitScriptFile = configParams.commit_filter.commitScriptFile;
+fiberWeightsFile = configParams.commit_filter.fiberWeightsFile;
+filteredConnectivityMatrixFile = configParams.commit_filter.filteredConnectivityMatrixFile;
+outputCommitDir = configParams.commit_filter.outputCommitDir;
+lambda = configParams.commit_filter.lambda;
+wmMaskFile = configParams.commit_filter.wmMaskFile;
+subjectDir = pwd;
+
+
+%% Convert bvals and bvecs to single scheme file
+
+% DWI.scheme has the following organization:
+% VERSION: BVECTOR
+% -0.000000	-0.000000	1.000000	0.000000
+% 0.971803	-0.194807	-0.132847	700.000000
+
+gtab = load_gtab(processedBvalsFile, processedBvecsFile, bValueZeroThreshold, bValueScalingTol);
+
+
+[~, ~] = mkdir(fileparts(schemeFile));
+fid = fopen(schemeFile, 'w+');
+
+if fid == -1
+     error('CATO:commit_filter:cannotCreateFile', 'Cannot create ''%s''.', schemeFile);
+end
+
+fprintf(fid, 'VERSION: BVECTOR\n');
+fprintf(fid, '%.6f\t%.6f\t%.6f\t%.6f\n', [gtab.bvecs gtab.bvals]');
+
+fclose(fid);
+
+
+%% Create white matter mask file
+
+segmentation = load_nifti(segmentationFile);
+
+whiteMatterMask = segmentation;
+whiteMatterMask.vol = zeros(size(segmentation.vol));
+whiteMatterMask.vol(ismember(segmentation.vol, startRegions)) = 1;
+
+save_nifti(whiteMatterMask, wmMaskFile);
+
+%% Reconstruct networks for each template and each method.
+for iTemplate = 1:length(templates)
+    thisTemplate = templates{iTemplate};
+    
+    for iMethod = 1:length(reconstructionMethods)
+        thisMethod = reconstructionMethods{iMethod};
+        
+        thisFiberFile =  strrep(fiberFile, ...
+            'METHOD', thisMethod);
+        thisFilteredFiberFile =  strrep(filteredFiberFile, ...
+            'METHOD', thisMethod);
+        thisFiberPropertiesFile =  strrep(strrep(fiberPropertiesFile, ...
+            'METHOD', thisMethod), ...
+            'TEMPLATE', thisTemplate);
+        thisFilteredFiberPropertiesFile =  strrep(strrep(filteredFiberPropertiesFile, ...
+            'METHOD', thisMethod), ...
+            'TEMPLATE', thisTemplate);        
+        thisRegionPropertiesFile = strrep(regionPropertiesFile, ...
+            'TEMPLATE', thisTemplate);        
+
+        fprintf('%s - %s\n', thisTemplate, upper(thisMethod));
+
+        % Open reconstructed fiber file.
+        % This script assumes that the fibercloud data is small enough to
+        % fit into memory, which is quite a liberal assumption.
+        fprintf('Load fibers...');
+        [fibers, headerFiberFile] = readTrk(thisFiberFile);
+        fprintf(' done\n');
```

Comment:
If this may be problematic we could wrap it in a try/catch statement?

Reply:
The `readTrk` function had indeed limited error handling, which could result in unexpected errors when reading the fiber file in this code. Fortunately, the error handling of `readTrk` has already been improved in commit f136c85 on the main branch.

I have updated this feature branch to include the aforementioned commit from the main branch and this update obviates the need for a try/catch statement.